src/backend/adapters/base-record.js

```diff
@@ -63,6 +63,7 @@ class BaseRecord {
     } catch (e) {
       if (e instanceof ValidationError) {
         this.errors = e.errors
+        this.storeParams(params)
```

Comment:
I would still put this above because we are not sure how update function will return params - I almost sure that in mongoose case they are not flattened. So after updating the record (without errors) you will have an unflatten params object.